lib/_tls_common.js

```diff
@@ -193,113 +244,113 @@ exports.createSecureContext = function createSecureContext(options) {
+  }
 
-  if (options.ecdhCurve === undefined)
-    c.context.setECDHCurve(tls.DEFAULT_ECDH_CURVE);
-  else if (options.ecdhCurve)
-    c.context.setECDHCurve(options.ecdhCurve);
+  validateString(ecdhCurve, 'options.ecdhCurve');
+  c.context.setECDHCurve(ecdhCurve);
 
-  if (options.dhparam) {
-    const warning = c.context.setDHParam(options.dhparam);
+  if (dhparam !== undefined) {
+    validateKeyOrCertOption('dhparam', dhparam);
+    const warning = c.context.setDHParam(dhparam);
     if (warning)
       process.emitWarning(warning, 'SecurityWarning');
   }
 
-  if (options.crl) {
-    if (ArrayIsArray(options.crl)) {
-      for (const crl of options.crl) {
-        c.context.addCRL(crl);
+  if (crl !== undefined) {
+    if (ArrayIsArray(crl)) {
+      for (const val of crl) {
+        validateKeyOrCertOption('crl', val);
+        c.context.addCRL(val);
       }
     } else {
-      c.context.addCRL(options.crl);
+      validateKeyOrCertOption('crl', crl);
+      c.context.addCRL(crl);
     }
   }
 
-  if (options.sessionIdContext) {
-    c.context.setSessionIdContext(options.sessionIdContext);
+  if (sessionIdContext !== undefined) {
+    validateString(sessionIdContext, 'options.sessionIdContext');
+    c.context.setSessionIdContext(sessionIdContext);
   }
 
-  if (options.pfx) {
-    if (!toBuf)
-      toBuf = require('internal/crypto/util').toBuf;
-
-    if (ArrayIsArray(options.pfx)) {
-      for (const pfx of options.pfx) {
-        const raw = pfx.buf ? pfx.buf : pfx;
-        const buf = toBuf(raw);
-        const passphrase = pfx.passphrase || options.passphrase;
-        if (passphrase) {
-          c.context.loadPKCS12(buf, toBuf(passphrase));
+  if (pfx !== undefined) {
+    if (ArrayIsArray(pfx)) {
+      for (const val of pfx) {
+        const raw = val.buf ? val.buf : val;
+        const pass = val.passphrase || passphrase;
+        if (pass !== undefined) {
+          c.context.loadPKCS12(toBuf(raw), toBuf(pass));
         } else {
-          c.context.loadPKCS12(buf);
+          c.context.loadPKCS12(toBuf(raw));
         }
       }
+    } else if (passphrase) {
+      c.context.loadPKCS12(toBuf(pfx), toBuf(passphrase));
     } else {
-      const buf = toBuf(options.pfx);
-      const passphrase = options.passphrase;
-      if (passphrase) {
-        c.context.loadPKCS12(buf, toBuf(passphrase));
-      } else {
-        c.context.loadPKCS12(buf);
-      }
+      c.context.loadPKCS12(toBuf(pfx));
     }
   }
 
   // Do not keep read/write buffers in free list for OpenSSL < 1.1.0. (For
   // OpenSSL 1.1.0, buffers are malloced and freed without the use of a
   // freelist.)
-  if (options.singleUse) {
+  if (singleUse) {
     c.singleUse = true;
     c.context.setFreeListLength(0);
   }
 
-  if (typeof options.clientCertEngine === 'string') {
-    if (c.context.setClientCertEngine)
-      c.context.setClientCertEngine(options.clientCertEngine);
-    else
+  if (clientCertEngine != null) {
```

Comment:
since this is semver-major, I'd stop accepting `null` here too (and make it consistent with the rest)

```suggestion
  if (clientCertEngine !== undefined) {
```